src/diffusers/models/transformer_2d.py

```diff
@@ -91,6 +91,7 @@ def __init__(
         upcast_attention: bool = False,
         norm_type: str = "layer_norm",
         norm_elementwise_affine: bool = True,
+        use_gated_attention: bool = False,
```

Comment:
```suggestion
        attention_type: str = "default",  # gated
```

When introducing new config variables let's make sure we can extend them going forward. Using a string-type variable would be nice